core/src/main/java/it/pagopa/selfcare/external_api/core/UserServiceImpl.java

```diff
@@ -136,39 +142,62 @@ public List<OnboardedInstitutionInfo> getOnboardedInstitutionsDetails(String use
     }
 
     @Override
-    public List<OnboardedInstitutionInfo> getOnboardedInstitutionsDetailsActive(String userId, String productId) {
-        List<UserInstitution> institutionsWithProductActive = userMsConnector.getUsersInstitutions(userId, null, null, null, null, Objects.isNull(productId) ? null : List.of(productId), null, null)
+    public List<OnboardedInstitutionResource> getOnboardedInstitutionsDetailsActive(String userId, String productId) {
+        List<UserInstitution> institutionsWithProductActive = userMsConnector.getUsersInstitutions(userId, null, null, null, null, Objects.isNull(productId) ? null : List.of(productId), null, List.of(ACTIVE.name()))
                 .stream()
                 .filter(item -> Objects.nonNull(item.getProducts()))
-                .filter(item -> item.getProducts().stream()
-                        .filter(product -> Objects.nonNull(product.getProductId()))
-                        .anyMatch(product -> product.getProductId().equals(productId) && RelationshipState.ACTIVE.name().equals(product.getStatus())))
-                .peek(item -> item.setProducts(item.getProducts().stream()
-                        .filter(product -> product.getProductId().equals(productId) && RelationshipState.ACTIVE.name().equals(product.getStatus()))
-                        .toList()))
+                .map(userInstitution -> filterProductList(userInstitution, productId))
```

Comment:
maybe could we avoid to filterProduct for ACTIVE again because we just filter above (line 146)